easybuild/easyblocks/g/gcc.py

```diff
@@ -368,15 +413,40 @@ def configure_step(self):
                     'val': os.getenv('LD_LIBRARY_PATH')
                 }
                 env.setvar('LD_LIBRARY_PATH', ld_lib_path)
-                extra_source = {1: "nvptx-tools", 2: "newlib"}[self.iter_idx]
-                extra_source_dirs = glob.glob(os.path.join(self.builddir, '%s-*' % extra_source))
-                if len(extra_source_dirs) != 1:
-                    raise EasyBuildError("Failed to isolate %s source dir" % extra_source)
-                if self.iter_idx == 1:
-                    # compile nvptx-tools
+                if self.build_stages[self.iter_idx] == NVPTX_TOOLS:
+                    # Configure NVPTX tools and build
+                    extra_source_dirs = glob.glob(os.path.join(self.builddir, 'nvptx-tools-*'))
+                    if not extra_source_dirs:
+                        raise EasyBuildError("Failed to isolate NVPTX source dir")
                     change_dir(extra_source_dirs[0])
-                else:  # self.iter_idx == 2
+                    return super(EB_GCC, self).configure_step()
+                elif self.build_stages[self.iter_idx] == AMD_LLVM:
+                    # Setup symlink from LLD source to 'builddir/lld' so that LLVM can find it
```

Comment:
This is a sizable extra chunk of code, maybe move it to a dedicated method?

Reply:
With the new reductions in code, do you still think moving this to a separate method makes sense? I'm leaning towards not moving the code since it doesn't make sense as a separate method and I think it will only lead to indirection when needing to change the code in the future.